### PR TITLE
fix: Remove to_graphql and make the interpreter runtime a requirement for older GraphQL versions

### PR DESCRIPTION
**Pull request by daemonsy**
# What's up? 

Trying to deal with the problem caused by `to_graphql` in https://github.com/Gusto/apollo-federation-ruby/issues/151. 

## The cause

GraphQL Ruby is deprecating the `to_graphql` method, slated for removal in 2.0. In 1.13, an argument was introduced `def to_graphql(silence_deprecation_warning: false)`. 

We have a `to_graphql` method in our codebase to add federation directives decorations. It is breaking because we defined it as a method with no arguments. 

Tests didn't break because we don't actually call this method in our specs. We rely on GraphQL ruby to call it in GraphQL versions where the **interpreter runtime is not the default.**

## Complications 

First thought was to add a spec to it, proxy arguments over. I'm used to proxying args `Ruby 2.7.x` style. It failed in our specs because the specs are running on `Ruby 2.6.x`. See this [article](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/). 

Given that Ruby 3.x is out, our keyword proxying should do the right thing. We also really didn't want to check for arity. 

## Let's just remove `to_graphql`

Specs fail if we remove the method, but why? The library requires the call to populate metadata on the legacy runtime. 

If we made the interpreter runtime a requirement on older versions of GraphQL where it wasn't the default, all specs pass. 

Given how old these versions are, I think this is the best solution. We're going to remove its use for 2.0 anyway.

### CODE REVIEW

File: example/products.rb [hunk 1]
```diff
@@ -58,6 +58,11 @@ def top_products(first:)
 end
 
 class ProductSchema < GraphQL::Schema
+  if Gem::Version.new(GraphQL::VERSION) < Gem::Version.new('1.12.0')
```
Comment (daemonsy) [inline example/products.rb:61]: All the other test schemas already have this code block, these two just happened to be missed 🤷‍♂️

**Comment by daemonsy** [comment]
I'd think we can remove support for GraphQL < 1.12 once we support 2.0. But I'd like to get some stats first if I can to confirm that they're not widely used by the community.

**Review by grxy — APPROVED** [review]
LGTM

**Pull request merged**